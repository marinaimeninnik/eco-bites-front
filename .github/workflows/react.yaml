name: Deploy to S3 and Invalidate CloudFront

on:
  push:
    branches:
      - main
      - CI-CD_pipeline
  workflow_dispatch:
  workflow_call:

jobs:
  deploy:
    permissions:
        id-token: write
        contents: read
    runs-on: ubuntu-latest

    steps:
    - name: Checkout
      uses: actions/checkout@v2

    - name: Configure AWS credentials
      uses: aws-actions/configure-aws-credentials@v4
      with:
        role-to-assume: ${{ vars.ROLE_ARN }}
        role-session-name: github-actions-labsadm
        aws-region: ${{ vars.AWS_REGION }}

    - name: Download artifact
      uses: actions/download-artifact@v4
      with:
        name: tf-outputs
        repository: marinaimeninnik/eco-bites-back
       
    - name: Load instance ID from file
      run: cat tf-outputs.env >> $GITHUB_ENV

    - name: Check variables
      run: echo 'bucket_name is ${{ env.FRONTEND_S3_NAME }}'

    - name: Build frontend
      run: npm run build

    # - name: Deploy to S3
    #   run: aws s3 sync ./build s3://${{ env.FRONTEND_S3_NAME }}
    # # env:
    # #     AWS_S3_BUCKET: ${{ secrets.AWS_S3_BUCKET }}

    # - name: Invalidate CloudFront Distribution
    #   uses: chetan/invalidate-cloudfront-action@v1
    #   env:
    #     DISTRIBUTION: ${{ env.CF_ID }}
    #     PATHS: '/*'
    #     #   AWS_REGION: 'us-west-2'
    #     #     AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
    #     #     AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}