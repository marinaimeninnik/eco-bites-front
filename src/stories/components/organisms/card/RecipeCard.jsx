import Link from '@mui/material/Link';
import Card from '@mui/material/Card';
import CardMedia from '@mui/material/CardMedia';
import CardContent from '@mui/material/CardContent';
import RecipeTitle from '../../atomic/text-elements/RecipeTitle';
import FavoriteRecipeIcon from '../../atomic/icons/FavoriteRecipeIcon';
import RecipeDuration from '../../molecules/text-elements/RecipeDuration';
import imagePlaceholder from '../../../assets/recipe-image-default.png';
import { recipeCardPropTypes } from '../../../prop-types/entities';

const RecipeCard = ({ recipe }) => {
    const {
        title,
        duration,
        imgUrl = imagePlaceholder,
        alt = 'Awaiting image',
        isFavorite = false,
        isVegan = false,
    } = recipe;

    return (
        <Link
            href="#"
            aria-label={`Click to see recipe for ${title}.`}
            style={{ textDecoration: 'none' }}
        >
            <Card>
                <CardMedia
                    component="img"
                    height="103"
                    image={imgUrl}
                    alt={alt}
                />
                {isFavorite && (
                    <FavoriteRecipeIcon title={title} selected={true} />
                )}
                <CardContent sx={{ mt: 2 }}>
                    <RecipeDuration
                        minutes={duration}
                        showClock={false}
                        variant="Paragraph/P12"
                    />
                    <RecipeTitle title={title} isVegan={isVegan} />
                </CardContent>
            </Card>
        </Link>
    );
};

RecipeCard.propTypes = recipeCardPropTypes;

export default RecipeCard;
