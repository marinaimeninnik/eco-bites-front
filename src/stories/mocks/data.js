export const image = 'https://source.unsplash.com/random';

export const recipeCardMockData = [
    {
        id: '00001',
        title: 'Veggie Curry',
        duration: 30,
        imgUrl: image,
        alt: 'veggie curry',
        dish: 'Main Course',
        diet: 'Vegetarian',
        isFavorite: false,
        isVegan: false,
    },
    {
        id: '00002',
        title: 'Poached Salmon with Jasmine Rice',
        duration: 90,
        imgUrl: image,
        alt: 'Poached Salmon',
        dish: 'Main Course',
        diet: 'Pescatarian',
        isFavorite: false,
        isVegan: false,
    },
    {
        id: '00003',
        title: 'Seitan Burger',
        duration: 70,
        imgUrl: image,
        alt: 'Seitan Burger',
        dish: 'Main Course',
        diet: 'Vegan',
        isFavorite: true,
        isVegan: true,
    },
    {
        id: '00004',
        title: 'Vegan Lentils Soup',
        duration: 30,
        imgUrl: image,
        alt: 'Vegan Lentils Soup',
        dish: 'Appetizers',
        diet: 'Vegan',
        isFavorite: true,
        isVegan: true,
    },
    {
        id: '00005',
        title: 'Stir-fried Broccoli',
        duration: 45,
        imgUrl: image,
        alt: 'Stir-fried Broccoli',
        dish: 'Main Course',
        diet: 'Vegan',
        isFavorite: false,
        isVegan: true,
    },
    {
        id: '00006',
        title: 'Grilled Asparagus',
        duration: 30,
        imgUrl: undefined,
        alt: undefined,
        dish: 'Side Dishes',
        diet: 'Vegan',
        isFavorite: false,
        isVegan: true,
    },
    {
        id: '00007',
        title: 'BBQ Chicken Wings',
        duration: 120,
        imgUrl: image,
        alt: 'BBQ Chicken Wings',
        dish: 'Main Course',
        diet: 'Flexitarian',
        isFavorite: true,
        isVegan: false,
    },
    {
        id: '00008',
        title: 'Lemon Tart',
        duration: 80,
        imgUrl: image,
        alt: 'Lemon Tart',
        dish: 'Desserts',
        diet: 'Vegetarian',
        isFavorite: true,
        isVegan: false,
    },
    {
        id: '00009',
        title: 'Chocolate Cake',
        duration: 90,
        imgUrl: image,
        alt: 'Chocolate Cake',
        dish: 'Desserts',
        diet: 'Vegetarian',
        isFavorite: true,
        isVegan: false,
    },
    {
        id: '00010',
        title: 'Berry Smoothie',
        duration: 10,
        imgUrl: image,
        alt: 'Berry Smoothie',
        dish: 'Beverages',
        diet: 'Vegan',
        isFavorite: true,
        isVegan: true,
    },
    {
        id: '00011',
        title: 'Recipe 11',
        duration: 30,
        imgUrl: image,
        alt: 'Recipe 11 Alt Text',
        dish: 'Main Course',
        diet: 'Mediterranean',
        isFavorite: false,
        isVegan: false,
    },
];

export default {
    recipeCardMockData,
};
