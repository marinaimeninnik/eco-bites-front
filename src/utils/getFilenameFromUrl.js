/**
 * Extracts the filename (pathname) from a URL.
 *
 * @param {string} urlString - The URL string to extract the filename from.
 *
 * @returns {string} - Returns the filename (pathname) from the URL without the
 *                    leading '/'. If URL is invalid or does not contain a pathname,
 *                    it returns an empty string.
 */
export default function getFilenameFromUrl(urlString) {
    let url = new URL(urlString);
    return url.pathname.substring(1);
}
