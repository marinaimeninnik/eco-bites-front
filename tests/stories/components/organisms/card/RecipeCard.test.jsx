import { render } from '@testing-library/react';
import RecipeCard from 'src/stories/components/organisms/card/RecipeCard';
import getFilenameFromUrl from 'src/utils/getFilenameFromUrl';
import imagePlaceholder from 'src/stories/assets/recipe-image-default.png';
import { recipeCardMockData, image } from 'src/stories/mocks/data';

describe('RecipeCard Component', () => {
    const testProps = recipeCardMockData[0];
    it('renders correctly', () => {
        const { asFragment } = render(<RecipeCard recipe={testProps} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders without crashing', () => {
        const { getByText } = render(<RecipeCard recipe={testProps} />);
        expect(getByText('Veggie Curry')).toBeInTheDocument();
    });

    it('shows FavoriteRecipeIcon when isFavorite is true', () => {
        const favoriteRecipe = {
            ...testProps,
            isFavorite: true,
        };
        const { queryByTestId } = render(
            <RecipeCard recipe={favoriteRecipe} />
        );
        expect(queryByTestId('favorite-icon')).toBeInTheDocument();
    });

    it('does not show FavoriteRecipeIcon when isFavorite is false', () => {
        const { queryByTestId } = render(
            <RecipeCard isFavorite={false} recipe={testProps} />
        );
        expect(queryByTestId('favorite-icon')).toBeNull();
    });

    it('renders img with correct src and alt attributes', () => {
        const { getByRole } = render(<RecipeCard recipe={testProps} />);
        const imgElement = getByRole('img');
        expect(imgElement.src).toBe(image);
        expect(imgElement.alt).toBe('veggie curry');
    });

    it('renders img with default src when no imgUrl prop is passed', () => {
        const withNoImage = {
            ...testProps,
            imgUrl: undefined,
            alt: undefined,
        };
        const { getByAltText } = render(<RecipeCard recipe={withNoImage} />);
        const imgElement = getByAltText('Awaiting image');
        expect(getFilenameFromUrl(imgElement.src)).toBe(imagePlaceholder);
    });
});
