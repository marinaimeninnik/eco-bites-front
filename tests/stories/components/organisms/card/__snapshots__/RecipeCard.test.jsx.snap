// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard Component renders correctly 1`] = `
<DocumentFragment>
  <a
    aria-label="Click to see recipe for Veggie Curry."
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
    href="#"
    style="text-decoration: none;"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <img
        alt="veggie curry"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        height="103"
        src="https://source.unsplash.com/random"
      />
      <div
        class="MuiCardContent-root css-1nyxmgy-MuiCardContent-root"
      >
        <div
          class="MuiStack-root css-ql8i53-MuiStack-root"
        >
          <time
            class="MuiTypography-root MuiTypography-Emphasis/E12 css-37jdci-MuiTypography-root"
            datetime="PT30M"
          >
            30m
          </time>
        </div>
        <h2
          class="MuiTypography-root MuiTypography-Emphasis/E14 css-mc2qb1-MuiTypography-root"
        >
          <span
            class="css-irdm38"
          >
            Veggie Curry
          </span>
        </h2>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
